pkg/applications/providers/source/helm.go

```diff
@@ -54,7 +54,7 @@ func (h HelmSource) DownloadSource(destination string) (string, error) {
 	}
 	defer util.CleanUpHelmTempDir(helmCacheDir, h.Log)
 
-	auth, err := util.HelmAuthFromCredentials(h.Ctx, h.SeedClient, path.Join(helmCacheDir, "reg-creg"), h.SecretNamespace, h.Source)
+	auth, err := util.HelmAuthFromCredentials(h.Ctx, h.SeedClient, path.Join(helmCacheDir, "reg-creg"), h.SecretNamespace, h.Source.Credentials)
```

Comment:
I have question regarding the two calls to util.HelmAuthFromCredentials. The other one being this here https://github.com/vgramer/kubermatic/blob/67826b8392a8ecde3e9898357eae5e3c31176d0f/pkg/applications/providers/template/helm.go#L68

I see that in one of the invocations we check if the credentials is set, while in the other we don't. Is there anything to it? 

I am thinking that in the other one we avoid unnecessary calls by checking first. Would we not want a nil-check on h.Source.Credentials here as well?

Reply:
the `util.HelmAuthFromCredentials` function takes a pointer to `*appskubermaticv1.HelmCredentials` and handle nil case. 
As` h.Source.Credentials`'s type is `*appskubermaticv1.HelmCredentials` we don't need to check if it is nil.

Reply:
Okay gotcha and in the other one we need the check so we can be sure that TemplateCredentials.HelmCredentials does not pani. Makes sense.